app/src/main/java/com/kdani/dagger2_11/di/module/ActivityModule.java

```diff
@@ -7,6 +7,10 @@
 import dagger.Module;
 import dagger.android.ContributesAndroidInjector;
 
+/**
+ * Created by KD on 11/5/17.
+ */
+
 @Module
 public abstract class ActivityModule {
 
```

Comment:
the moment i add another activityModule here i get the same error 

```java
    @PerActivity
    @ContributesAndroidInjector(modules = ABActivityModule.class)
    abstract ABActivity abActivity();
```
_ABActivityModule.class_
```java
@Module
public abstract class ABActivityModule {
}
```

Reply:
I tried this too but it results in same
```java
@Module(includes = BaseActivityModule.class)
public abstract class ABActivityModule {
}
```

Reply:
You still need the binding in your `ABCActivityModule`:

 `@PerActivity @Binds abstract BaseActivity baseActivity(ABCActivity abcActivity);`

 otherwise Dagger 2 doesn't know that `ABCActivity` is an instance of `BaseActivity`

Reply:
I see to pretty much any activity that extends baseActivity will have to have it's separate module and a necessary method like this below

```java
@Module(includes = BaseActivityModule.class)
public abstract class ABActivityModule {
    @PerActivity
    @Binds
    abstract BaseActivity baseActivity(ABActivity mainActivity);
}
```

and as you said all of this is bcz Dagger doesn't know ABCActivity is the instance of BaseActivity

Reply:
Thanks a ton :1st_place_medal:  for putting your precious time towards my problem. It's kinda odd that I landed into this scenario but i'm happy that i was able to make out of it with your help. At least now I have some more familiarity with dagger 2.11